### ISSUE TITLE
Make sync --exclude apply to both client and receiving sides

### ISSUE THREAD

**Issue by mentalblock** [s3tools/s3cmd#109]
When a file does not exist on the source but exists on the remote the command `s3cmd sync --delete-removed --exclude "testfile" . s3://BUCKET` should not delete the remote file `testfile` if it does not exist locally.

In other words, the `--exclude` option should apply to the transfers of files and be direction-agnostic. Please see the documentation on the `--delete` option for `rsync`. The `rsync` command supports include/exclude modifiers which allow flexibility in determining whether they affect the client side or receiving side.

Please make the default similar to `rsync` where the `--exclude` option applies to both the client and receiving side.

An example diff may look like the code below if we were to apply this to the `cmd_sync_local2remote` method.

```
diff --git a/s3cmd b/s3cmd
index aeb7706..b1ee3e6 100755
--- a/s3cmd
+++ b/s3cmd
@@ -864,7 +864,8 @@ def cmd_sync_local2remote(args):

     info(u"Found %d local files, %d remote files" % (local_count, remote_count))

-    local_list, exclude_list = filter_exclude_include(local_list)
+    local_list, exclude_list_local = filter_exclude_include(local_list)
+    remote_list, exclude_list_remote = _filelist_filter_exclude_include(remote_list)

     if single_file_local and len(local_list) == 1 and len(remote_list) == 1:
         ## Make remote_key same as local_key for comparison if we're dealing with only one file
@@ -889,8 +890,9 @@ def cmd_sync_local2remote(args):
             for key in local_list:
                 local_list[key]['remote_uri'] = unicodise(destination_base + key)

+    exclude_list_final = dict(exclude_list_local.items() + exclude_list_remote.items())
     if cfg.dry_run:
-        for key in exclude_list:
+        for key in exclude_list_final:
             output(u"exclude: %s" % unicodise(key))
         if cfg.delete_removed:
             for key in remote_list:
```

**Comment by kribblo** [comment]
Should this be closed, since #152 is closed?

**Comment by fviard** [comment]
@kribblo : Right, thanks for the notice!
